site/en/guide/autodiff.ipynb

```diff
@@ -895,9 +895,9 @@
         "\n",
         "State stops gradients. When you read from a stateful object the tape can only see the current state, not the history that lead to it.\n",
         "\n",
-        "A `tf.Tensor` is immutable. You can't change a tensor once it's created. It has a _value_, but no _state_. All the operations discussed so far are also stateless: The output of a `tf.matmul` only depends on its inputs.\n",
+        "A `tf.Tensor` is immutable. You can't change a tensor once it's created. It has a _value_, but no _state_. All the operations discussed so far are also stateless: the output of a `tf.matmul` only depends on its inputs.\n",
         "\n",
-        "A `tf.Variable` has internal state, its value. When you use the variable, the state is read. It's normal to calculate a gradient with respect to a variable, but the variable's state blocks gradient calculations from going farther back. For example:\n"
+        "A `tf.Variable` has internal state — its value. When you use the variable, the state is read. It's normal to calculate a gradient with respect to a variable, but the variable's state blocks gradient calculations from going farther back. For example:\n"
```

Comment:
```suggestion
        "A `tf.Variable` has internal state—its value. When you use the variable, the state is read. It's normal to calculate a gradient with respect to a variable, but the variable's state blocks gradient calculations from going farther back. For example:\n"
```

Reply:
Applying this logic—using an em dash with no spaces on either side—across the docs in this PR 👍 Thanks @lamberta